src/arbitraries/MappedArbitrary.ts

```diff
@@ -1,9 +1,16 @@
-import {FluentPick} from './types'
+import {FluentPick, MappedArbitraryConfig} from './types'
 import {Arbitrary} from './internal'
 
 export class MappedArbitrary<A, B> extends Arbitrary<B> {
-  constructor(public readonly baseArbitrary: Arbitrary<A>, public readonly f: (a: A) => B) {
+  constructor(
+    public readonly baseArbitrary: Arbitrary<A>,
+    public readonly f: (a: A) => B,
+    public readonly config?: MappedArbitraryConfig<A,B>
+  ) {
     super()
+
+    if (this.config && this.config.customCanGenerate)
```

Comment:
nit: this can be a one-liner. 

```suggestion
    this.canGenerate = this.config?.customCanGenerate || this.canGenerate
```

Reply:
Yeah I know but I was getting a problem in doing that, that now it is fixed. Already added your suggestion!

Reply:
Sometimes directly writing a _canGenerate_ function can be much easier than going the _inverse_ route.